terraform/transform_attach_schema.go

```diff
@@ -59,8 +60,16 @@ func (t *AttachSchemaTransformer) Transform(g *Graph) error {
 			mode := addr.Resource.Mode
 			typeName := addr.Resource.Type
 			providerAddr, _ := tv.ProvidedBy()
+			var providerFqn addrs.Provider
 
-			schema, version := t.Schemas.ResourceTypeConfig(providerAddr.Provider, mode, typeName)
+			switch providerAddr.(type) {
```

Comment:
you can assign a value in a type switch, e.g.

`switch p := providerAddr.(type) {`

Then use it in each case with the correct type:

`providerFqn = p.Provider`

Reply:
oh hey thanks! That's so much more readable!